Content.Server/Speech/EntitySystems/FrenchAccentSystem.cs

```diff
@@ -0,0 +1,38 @@
+using Content.Server.Speech.Components;
+using System.Text.RegularExpressions;
+
+namespace Content.Server.Speech.EntitySystems;
+
+public sealed class FrenchAccentSystem : EntitySystem
+{
+    [Dependency] private readonly ReplacementAccentSystem _replacement = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<FrenchAccentComponent, AccentGetEvent>(OnAccentGet);
+    }
+
+    // converts left word when typed into the right word. For example typing you becomes ye.
+    public string Accentuate(string message, FrenchAccentComponent component)
+    {
+        var msg = message;
+
+        msg = _replacement.ApplyReplacements(msg, "french");
+
+        // replaces th with dz 
+        msg = Regex.Replace(msg, @"th", "dz", RegexOptions.IgnoreCase);
+
+        // removes the letter h from the start of words.
+        msg = Regex.Replace(msg, @"(?<!\w)[h]", "", RegexOptions.IgnoreCase);
```

Comment:
Suggestion: insert an apostrophe instead of deleting the letter, so one can tell a letter was omitted.

English as a large number of h-∅ minimal pairs (words that differ only in the presence of 'h'). This might make it ard to guess the original intent.

Reply:
this is now an apostrophe instead of deleting letter.